flake.nix

```diff
@@ -13,57 +12,63 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
+    inherit (nixpkgs) lib;
+    systems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    eachSystem = lib.genAttrs systems;
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
         inherit system;
         overlays = [(import rust-overlay)];
-      };
+      });
+    gitRev = self.rev or self.dirtyRev or null;
+  in {
+    packages = eachSystem (system: {
+      helix = pkgsFor.${system}.callPackage ./default.nix {inherit gitRev;};
```

Comment:
After changing this to use `mapAttrs op pkgsFor`, can you please `inherit (pkgs) helix` so that the overlay is the source of truth for the `callPackage` invocation?

Reply:
Worked for me.

I actually would like to give a little nuance to the pattern.

In this particular position, where `packages` exposes those inherited from our instantiation of `nixpkgs`, I do like to explicitly say `inherit (pkgsFor.${system}) helix` as it makes things very clear. For the other outputs it doesn't matter so much; it is assumed that any `pkgs` is ours.

Then again, I change my mind every other flake that I touch, because by my own reasoning, every `pkgs` is ours, so it's permissible even in the body of `packages` output. I'll leave it up to you.

Reply:
I'm not sure why it works in the first place, even with the direct `= callPackage` in `packages`. The package function does not take a `buildType`, that is an argument to `buildRustPackage` and must be overridden by `overrideAttrs`.

If it is desired to construct debug builds with `.override`, add it as an argument and `inherit buildType` in `buildRustPackage`.